src/main/java/mate.academy/lib/Injector.java

```diff
@@ -8,6 +19,60 @@ public static Injector getInjector() {
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Object clazzImplementationInstance = null;
+        Class<?> clazz = findImplementation(interfaceClazz);
+
+        if (!clazz.isAnnotationPresent(Component.class)) {
+            throw new RuntimeException(
+                    "Class " + clazz.getName() + " is not annotated with @Component"
+            );
+        }
+
+        Field[] declaredFields = clazz.getDeclaredFields();
+        for (Field field : declaredFields) {
+            if (field.isAnnotationPresent(Inject.class)) {
+                Object fieldInstance = getInstance(field.getType());
+
+                clazzImplementationInstance = createNewInstance(clazz);
```

Comment:
There is a potential issue here: `clazzImplementationInstance` is being assigned a new instance inside the loop for each field annotated with `@Inject`. This means if there are multiple fields with `@Inject`, the instance will be recreated multiple times, which is not the intended behavior. Consider moving the instantiation outside the loop to ensure the class is instantiated only once.